.github/workflows/build.yml

```diff
@@ -17,27 +17,46 @@ jobs:
       - name: Checkout sources
         uses: actions/checkout@v2
       - uses: actions/cache@v2
-        id: cache-id
+        id: build-linux-install-cache-id
         with:
-          path: ${{ runner.temp }}/cache-directory
-          key: ${{ runner.os }}-cache-key-v1.9
+          path: ${{ runner.temp }}/cache-build-linux-install
+          key: build-linux-install-cache-key-v1.0
       - uses: airvzxf/cache-anything-new-action@v1.0
         with:
           script: 'install-build.sh'
-          is_cached: ${{ steps.cache-id.outputs.cache-hit }}
-          cache: ${{ runner.temp }}/cache-directory
+          is_cached: ${{ steps.build-linux-install-cache-id.outputs.cache-hit }}
+          cache: ${{ runner.temp }}/cache-build-linux-install
           snapshot: '/'
           exclude: '/boot /data /dev /mnt /proc /run /sys'
+      - name: Prepare ccache timestamp
+        id: build-linux-ccache-timestamp
+        shell: cmake -P {0}
+        run: |
+          string(TIMESTAMP current_date "%Y-%m-%d-%H;%M;%S" UTC)
+          message("::set-output name=timestamp::${current_date}")
+      - name: ccache files
+        uses: actions/cache@v1.1.0
+        with:
+          path: ${{ runner.temp }}/ccache
+          key: build-linux-ccache-${{ steps.build-linux-ccache-timestamp.outputs.timestamp }}
```

Comment:
The timestamp for a cache looks strange to me. I don't understand how it could be reused.

I am not sure what is really cached, so maybe I will tell you some nonsense. 

In general, I expect 2 cache levels. 

1) Dependencies. Key is md5 of file/files with dependencies.  This thing is not changed often, so it could be reused between builds.  
2) Freeorion generated cache.  We can use a commit as a key (this will allow us to reuse cache during actions run on the same commit).  A more elegant way is to use the last commit that touches C++/build files. In that case, we could reuse cache for PRs that change only the content of `default` folder.  The best way is to cache modules separately.

Reply:
As I understand it uses two caches with different action version to cache dependencies and ccache. And `restore-key` allows to use ccache with different timestamp.

Reply:
It's based on https://github.com/freeorion/freeorion/pull/3270 and https://github.com/airvzxf/cache-anything-new-action/issues/1